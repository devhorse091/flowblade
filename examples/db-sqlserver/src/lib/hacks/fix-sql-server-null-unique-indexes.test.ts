import { fixSqlServerNullUniqueIndexes } from './fix-sql-server-null-unique-indexes';

describe('hack', () => {
  it('should fix unique indexes', () => {
    const ddl = `
      CREATE TABLE [common].[currency_i18n2] (
          [id] INT NOT NULL IDENTITY(1,1),
          [currency_id] SMALLINT NOT NULL,
          [locale] VARCHAR(5) NOT NULL,
          [name] VARCHAR(40) NOT NULL,
          [description] VARCHAR(40) NULL,
          [barcode_ean13] CHAR(13),
          [test] VARCHAR(40) NOT NULL,          
          CONSTRAINT [currency_i18n2_pkey] PRIMARY KEY CLUSTERED ([id]),          
          CONSTRAINT [currency_i18n2_name_key] UNIQUE NONCLUSTERED ([name]),
          CONSTRAINT [currency_i18n2_description_key] UNIQUE NONCLUSTERED ([description]),
          CONSTRAINT [currency_i18n2_barcode_ean13] UNIQUE NONCLUSTERED ([barcode_ean13]),
          CONSTRAINT [currency_i18n2_description_test_key] UNIQUE NONCLUSTERED ([description],[test]),
          CONSTRAINT [currency_i18n2_currency_id_locale_key] UNIQUE NONCLUSTERED ([currency_id],[locale]),
        ); 

      `;
    const fixed = fixSqlServerNullUniqueIndexes([ddl]);
    expect(fixed).toMatchSnapshot();
  });
});
