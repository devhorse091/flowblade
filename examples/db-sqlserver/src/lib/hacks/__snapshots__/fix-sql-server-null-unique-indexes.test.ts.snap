// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`hack > should fix unique indexes 1`] = `
[
  "CREATE TABLE [common].[currency_i18n2] (
          [id] INT NOT NULL IDENTITY(1,1),
          [currency_id] SMALLINT NOT NULL,
          [locale] VARCHAR(5) NOT NULL,
          [name] VARCHAR(40) NOT NULL,
          [description] VARCHAR(40) NULL,
          [barcode_ean13] CHAR(13),
          [test] VARCHAR(40) NOT NULL,          
          CONSTRAINT [currency_i18n2_pkey] PRIMARY KEY CLUSTERED ([id]),          
          CONSTRAINT [currency_i18n2_name_key] UNIQUE NONCLUSTERED ([name]),
                                        CONSTRAINT [currency_i18n2_currency_id_locale_key] UNIQUE NONCLUSTERED ([currency_id],[locale]),
        );",
  "CREATE UNIQUE NONCLUSTERED INDEX currency_i18n2_description_key
              ON [common].[currency_i18n2] ([description])
              WHERE [description] IS NOT NULL            
            ",
  "CREATE UNIQUE NONCLUSTERED INDEX currency_i18n2_barcode_ean13
              ON [common].[currency_i18n2] ([barcode_ean13])
              WHERE [barcode_ean13] IS NOT NULL            
            ",
  "CREATE UNIQUE NONCLUSTERED INDEX currency_i18n2_description_test_key
              ON [common].[currency_i18n2] ([description],[test])
              WHERE [description] IS NOT NULL AND [test] IS NOT NULL            
            ",
]
`;
